SG14/algorithm_ext.h

```diff
@@ -80,52 +78,80 @@ namespace stdext
 	template<class BidirIt, class UnaryPredicate>
 	BidirIt unstable_remove_if(BidirIt first, BidirIt last, UnaryPredicate p)
 	{
-		while (1) {
-			while ((first != last) && p(*first)) {
+		while (true) {
+			// Find the first instance of "p"...
+			while (true) {
+				if (first == last) {
+					return first;
+				}
+				if (p(*first)) {
+					break;
+				}
 				++first;
 			}
-			if (first == last--) break;
```

Comment:
@TBBle: This isn't about `end()--`; it's about `begin()--` (in the case when `last == begin()`).